### ISSUE TITLE
🐛 [firebase_auth] Crashes after moving onto Pigeon

### ISSUE THREAD

**Issue by andreystavitsky** [firebase/flutterfire#11347]
```
E/AndroidRuntime( 5199): java.lang.IllegalStateException: Reply already submitted
E/AndroidRuntime( 5199): 	at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:435)
E/AndroidRuntime( 5199): 	at io.flutter.plugin.common.BasicMessageChannel$IncomingMessageHandler$1.reply(BasicMessageChannel.java:222)
E/AndroidRuntime( 5199): 	at io.flutter.plugins.firebase.auth.GeneratedAndroidFirebaseAuth$FirebaseAuthUserHostApi$2.success(GeneratedAndroidFirebaseAuth.java:3247)
E/AndroidRuntime( 5199): 	at io.flutter.plugins.firebase.auth.GeneratedAndroidFirebaseAuth$FirebaseAuthUserHostApi$2.success(GeneratedAndroidFirebaseAuth.java:3244)
E/AndroidRuntime( 5199): 	at io.flutter.plugins.firebase.auth.FlutterFirebaseAuthUser.lambda$getIdToken$1(FlutterFirebaseAuthUser.java:88)
E/AndroidRuntime( 5199): 	at io.flutter.plugins.firebase.auth.FlutterFirebaseAuthUser$$ExternalSyntheticLambda15.onComplete(Unknown Source:2)
E/AndroidRuntime( 5199): 	at com.google.android.gms.tasks.zzi.run(com.google.android.gms:play-services-tasks@@18.0.2:1)
E/AndroidRuntime( 5199): 	at android.os.Handler.handleCallback(Handler.java:942)
E/AndroidRuntime( 5199): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime( 5199): 	at android.os.Looper.loopOnce(Looper.java:201)
```

```
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): Failed to handle message
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): java.lang.IllegalStateException: Task is not yet complete
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at com.google.android.gms.common.internal.Preconditions.checkState(com.google.android.gms:play-services-basement@@18.2.0:2)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at com.google.android.gms.tasks.zzw.zzf(com.google.android.gms:play-services-tasks@@18.0.2:1)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at com.google.android.gms.tasks.zzw.getResult(com.google.android.gms:play-services-tasks@@18.0.2:1)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at io.flutter.plugins.firebase.auth.FlutterFirebaseAuthUser.getIdToken(FlutterFirebaseAuthUser.java:95)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at io.flutter.plugins.firebase.auth.GeneratedAndroidFirebaseAuth$FirebaseAuthUserHostApi$-CC.lambda$setup$1(GeneratedAndroidFirebaseAuth.java:3256)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at io.flutter.plugins.firebase.auth.GeneratedAndroidFirebaseAuth$FirebaseAuthUserHostApi$$ExternalSyntheticLambda5.onMessage(Unknown Source:2)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at io.flutter.plugin.common.BasicMessageChannel$IncomingMessageHandler.onMessage(BasicMessageChannel.java:217)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:295)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$io-flutter-embedding-engine-dart-DartMessenger(DartMessenger.java:322)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at android.os.Handler.handleCallback(Handler.java:942)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at android.os.Looper.loopOnce(Looper.java:201)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at android.os.Looper.loop(Looper.java:288)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at android.app.ActivityThread.main(ActivityThread.java:7872)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at java.lang.reflect.Method.invoke(Native Method)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
E/BasicMessageChannel#dev.flutter.pigeon.FirebaseAuthUserHostApi.getIdToken( 5199): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```

```
[$AppUserRemoteAdapter] [firebase_auth/channel-error] Unable to establish connection on channel.
[$AppUserRemoteAdapter] #0      FirebaseAuthUserHostApi.getIdToken
messages.pigeon.dart:1490
                          <asynchronous suspension>
#1      MethodChannelUser.getIdToken
method_channel_user.dart:43
                          <asynchronous suspension>
```

**Comment by mashegoindustries** [comment]
Same issue,

For my use case, The line below crushes the App. try catch wont save me.

IdTokenResult idTokenResult = await _firebaseAuth.currentUser!.getIdTokenResult(true);



❯ flutter doctor -v
[!] Flutter (Channel stable, 3.10.5, on macOS 13.4.1 22F770820d darwin-x64, locale en-ZA)
    • Flutter version 3.10.5 on channel stable at /Users/***/Documents/flutter
    ! Warning: `dart` on your path resolves to /usr/local/Cellar/dart/2.8.4/libexec/bin/dart, which is not inside your current Flutter SDK
      checkout at /Users/***/Documents/flutter. Consider adding /Users/***/Documents/flutter/bin to the front of
      your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 796c8ef792 (6 weeks ago), 2023-06-13 15:51:02 -0700
    • Engine revision 45f6e00911
    • Dart version 3.0.5
    • DevTools version 2.23.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update
      checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/***/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] IntelliJ IDEA Community Edition (version 2023.1.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.80.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.68.0

[✓] Connected device (3 available)
    • sdk gphone x86 64 (mobile) • emulator-5554 • android-x64    • Android 13 (API 33) (emulator)
    • macOS (desktop)            • macos         • darwin-x64     • macOS 13.4.1 22F770820d darwin-x64
    • Chrome (web)               • chrome        • web-javascript • Google Chrome 114.0.5735.198

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
❯ clear
❯ flutter doctor -v
[!] Flutter (Channel stable, 3.10.5, on macOS 13.4.1 22F770820d darwin-x64, locale en-ZA)
    • Flutter version 3.10.5 on channel stable at /Users/****/Documents/flutter
    ! Warning: `dart` on your path resolves to /usr/local/Cellar/dart/2.8.4/libexec/bin/dart, which is not inside your current Flutter SDK
      checkout at /Users/****/Documents/flutter. Consider adding /Users/****/Documents/flutter/bin to the front of
      your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 796c8ef792 (6 weeks ago), 2023-06-13 15:51:02 -0700
    • Engine revision 45f6e00911
    • Dart version 3.0.5
    • DevTools version 2.23.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update
      checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/****/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] IntelliJ IDEA Community Edition (version 2023.1.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.80.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.68.0

[✓] Connected device (3 available)
    • sdk gphone x86 64 (mobile) • emulator-5554 • android-x64    • Android 13 (API 33) (emulator)
    • macOS (desktop)            • macos         • darwin-x64     • macOS 13.4.1 22F770820d darwin-x64
    • Chrome (web)               • chrome        • web-javascript • Google Chrome 114.0.5735.198

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.

**Comment by darshankawar** [comment]
Closing as duplicate of https://github.com/firebase/flutterfire/issues/11297.
